### PR TITLE
blas/gonum: add Zgerc

### PR DESCRIPTION
**Pull request by vladimir-ch**

### CODE REVIEW

File: blas/gonum/gonum.go [hunk 1]
```diff
@@ -74,3 +74,8 @@ func blocks(dim, bsize int) int {
 func dcabs1(z complex128) float64 {
 	return math.Abs(real(z)) + math.Abs(imag(z))
 }
+
+// conj returns the complex conjugate of x.
+func conj(z complex128) complex128 {
```
Comment (kortschak) [inline blas/gonum/gonum.go]: This exists in `cmplx` (and `cmplx64`).

File: blas/gonum/level2cmplx128.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+// Zgerc performs the rank-one operation
+//  A += alpha * x * y^H
+// where A is an m×n dense matrix, alpha is a scalar, x is an m element vector,
+// and y is an n element vector.
+func (Implementation) Zgerc(m, n int, alpha complex128, x []complex128, incX int, y []complex128, incY int, a []complex128, lda int) {
+	checkZMatrix('A', m, n, a, lda)
+	checkZVector('x', m, x, incX)
+	checkZVector('y', n, y, incY)
+
+	// Quick return if possible
```
Comment (kortschak) [inline blas/gonum/level2cmplx128.go]: Dot, and verb.

File: blas/gonum/gonum.go (hunk 1 above)
Comment (vladimir-ch) [inline blas/gonum/gonum.go]: I know, I thought that having it here might help with inlining. Anyway, removed and call cmplx version instead.

File: blas/gonum/level2cmplx128.go (hunk 2 above)
Comment (vladimir-ch) [inline blas/gonum/level2cmplx128.go]: Removed entirely.

File: blas/gonum/gonum.go (hunk 1 above)
Comment (kortschak) [inline blas/gonum/gonum.go]: AFAIU inlining happens across package boundaries.

Comment (btracey) [inline blas/gonum/gonum.go]: Is there any argument for legibility in keeping the call small?

**Comment by kortschak** [comment]
cmplx.Conj is pretty clear IMO.

**Comment by vladimir-ch** [comment]
Could you please re-approve? It's always the code generation. This time there are some changes also in `dot`, you might want to check, @kortschak .

**Comment by vladimir-ch** [comment]
It still says `The command "${TRAVIS_BUILD_DIR}/.travis/check-generate.sh" exited with 1` but I don't see any error. On my machine it exits with 0.

**Comment by kortschak** [comment]
We pin gocc at `2454fdc`, so you need to do the same. To get the generated code to be correct, you can run the following in the gonum root.

```
./.travis/install-gocc.sh 2454fdc
./.travis/check-generate.sh
```

**Comment by vladimir-ch** [comment]
Getting the Travis build to pass is a pain this time. The 1.7 job fails with this:
```
gocc dot.bnf
fatal error: unexpected signal during runtime execution
[signal SIGSEGV: segmentation violation code=0x1 addr=0xc41fff7fff pc=0x4134ae]

goroutine 1 [running]:
runtime.throw(0x6668c5, 0x2a)
	/home/travis/.gimme/versions/go1.7.6.linux.amd64/src/runtime/panic.go:566 +0x95 fp=0xc42009daa0 sp=0xc42009da80
runtime.sigpanic()
	/home/travis/.gimme/versions/go1.7.6.linux.amd64/src/runtime/sigpanic_unix.go:12 +0x2cc fp=0xc42009daf8 sp=0xc42009daa0
runtime.heapBitsSetType(0xc420060700, 0x680, 0x640, 0x621300)
	/home/travis/.gimme/versions/go1.7.6.linux.amd64/src/runtime/mbitmap.go:1266 +0x39e fp=0xc42009dbb8 sp=0xc42009daf8
runtime.mallocgc(0x680, 0x621300, 0x1, 0xc4200bc000)
	/home/travis/.gimme/versions/go1.7.6.linux.amd64/src/runtime/malloc.go:690 +0x5ba fp=0xc42009dc58 sp=0xc42009dbb8
runtime.makeslice(0x621300, 0x0, 0x64, 0xc4200bc000, 0x0, 0x64)
	/home/travis/.gimme/versions/go1.7.6.linux.amd64/src/runtime/slice.go:57 +0x7b fp=0xc42009dcb0 sp=0xc42009dc58
main.main()
	/home/travis/gopath/src/github.com/goccmack/gocc/main.go:80 +0x2b8 fp=0xc42009df38 sp=0xc42009dcb0
runtime.main()
	/home/travis/.gimme/versions/go1.7.6.linux.amd64/src/runtime/proc.go:183 +0x1f4 fp=0xc42009df90 sp=0xc42009df38
runtime.goexit()
	/home/travis/.gimme/versions/go1.7.6.linux.amd64/src/runtime/asm_amd64.s:2086 +0x1 fp=0xc42009df98 sp=0xc42009df90
make: *** [gen] Error 2
graph/formats/dot/dot.go:11: running "./makeinternal.bash": exit status 2

travis_time:end:1296595e:start=1502448527988823137,finish=1502448532193989566,duration=4205166429
[0K
[31;1mThe command "${TRAVIS_BUILD_DIR}/.travis/check-generate.sh" exited with 1.[0m

Done. Your build exited with 1.
```
I copy it here for reference, I'm going to restart the job.

**Comment by kortschak** [comment]
@mewmew, you might be interested in the error above.

**Pull request merged**